client/lib/plugins/wporg-data/test/list-store.js

```diff
@@ -1,279 +1,258 @@
-			var newPlugins = PluginsListsStore.getShortList( 'new' );
-			assert.equal( newPlugins.list.length, 0 );
+	describe( 'short lists', () => {
+		it( 'should be empty if the list has not been fetched yet', () => {
+			const newPlugins = PluginsListsStore.getShortList( 'new' );
+			assert.lengthOf( newPlugins.list, 0 );
 		} );
 
-		it( 'should return a list of plugins if the list has been fetched already', function() {
-			var newPlugins;
+		it( 'should return a list of plugins if the list has been fetched already', () => {
+			let newPlugins;
 			Dispatcher.handleServerAction( actionsData.fetchedNewPluginsList );
 			newPlugins = PluginsListsStore.getShortList( 'new' );
 			assert.isArray( newPlugins.list );
-			assert.equal( newPlugins.list.length, 1 );
+			assert.lengthOf( newPlugins.list, 1 );
 		} );
 
-		it( 'should not read from localStorage if the category isn\'t in the to-be-cached list', function() {
+		it( 'should not read from localStorage if the category isn\'t in the to-be-cached list', () => {
 			Dispatcher.handleServerAction( actionsData.fetchedNewPluginsList );
 			resetListsStore();
 			PluginsListsStore.getShortList( 'new' );
-			assert.equal( localStorageMock.getActivity().get, 0 );
+			assert.isFalse( localStorageSpies.get.called );
 		} );
 
-		it( 'should not store on localStorage if the category isn\'t in the to-be-cached list', function() {
+		it( 'should not store on localStorage if the category isn\'t in the to-be-cached list', () => {
 			Dispatcher.handleServerAction( actionsData.fetchedNewPluginsList );
-			assert.equal( localStorageMock.getActivity().set, 0 );
+			assert.isFalse( localStorageSpies.set.called );
 		} );
 
-		it( 'should try to read from localStorage if the category is in the to-be-cached list', function() {
+		it( 'should try to read from localStorage if the category is in the to-be-cached list', () => {
 			Dispatcher.handleServerAction( actionsData.fetchedPopularPluginsList );
 			resetListsStore();
 			PluginsListsStore.getShortList( 'popular' );
-			assert.notEqual( localStorageMock.getActivity().get, 0 );
+			assert.isTrue( localStorageSpies.get.called );
 		} );
 
-		it( 'should not be stored on localStorage if the category isn\'t in the to-be-cached list', function() {
+		it( 'should not be stored on localStorage if the category isn\'t in the to-be-cached list', () => {
 			Dispatcher.handleServerAction( actionsData.fetchedPopularPluginsList );
 			resetListsStore();
 			PluginsListsStore.getShortList( 'popular' );
-			assert.equal( localStorageMock.getActivity().set, 1 );
+			assert.isTrue( localStorageSpies.set.called );
 		} );
 
-		it( 'should not fetch from wporg if the category list is already in storage and is not yet after its TTL', function() {
-			Dispatcher.handleServerAction( actionsData.fetchedPopularPluginsList );
+		it( 'should not fetch from wporg if the category list is already in storage and is not yet after its TTL', () => {
 			resetListsStore();
```

Comment:
I'm not sure if the previous mock setup was incorrect but after converting to spies `fetchPluginsList` was being called if `resetListsStore` was called after the dispatch server action. That does seem to be the correct behavior where the `RESET_WPORG_PLUGINS_LIST` event clears out the storage.